bin/s2nd.c

```diff
@@ -355,6 +361,15 @@ int handle_connection(int fd, struct s2n_config *config, struct conn_settings se
         GUARD_RETURN(s2n_connection_use_corked_io(conn), "Error setting corked io");
     }
 
+    for (size_t i = 0; i < settings.psk_list_len; i++) {
+        struct s2n_psk *psk = s2n_external_psk_new();
+        GUARD_EXIT_NULL(psk);
+        GUARD_EXIT(s2n_setup_external_psk(&psk, settings.psk_optarg_list[i]), "Error setting external PSK parameters\n");
+        GUARD_EXIT(s2n_connection_append_psk(conn, psk), "Error appending psk to the connection\n");
+        GUARD_EXIT(s2n_psk_free(&psk), "Error freeing psk\n");
+        free(settings.psk_optarg_list[i]);
+    }
```

Comment:
Now that you're doing this outside of the optarg processing, couldn't you move it all to the common method and avoid this duplication?

Reply:
`s2n_setup_external_psk` setups only one PSK, this one setups and appends all the PSKs in the list. This function was just 2-3 lines so I didn`t see the need to add a new helper for it.